File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x<0) {
            return false;
        }
        int n = x;
        int revNum = 0;
        while(n>0) {
            int last_d = n%10;
            revNum = revNum*10 + last_d;
            n = n/10;
        }
        if(revNum == x) {
            return true;
        }
        else {
            return false;
        }
    }
}

File: 9-palindrome-number/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        rev = 0
        original = x
        sign = 1
        if x < 0:
            return False

        while x > 0:
            digit = x % 10
            rev = rev * 10 + digit
            x = x // 10

        if rev == original:
            return True
        else:
            return False